app/src/main/java/msi/paria/trainingapp/pages/main/MainActivity.kt

```diff
@@ -0,0 +1,32 @@
+package msi.paria.trainingapp.pages.main
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import msi.paria.trainingapp.pages.main.state.MainPageState
+import msi.paria.trainingapp.pages.main.view.MainPageView
+
+
+class MainActivity : ComponentActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val samplePageState = mutableStateOf(MainPageState())
```

Comment:
You don't need any state or data class. just remove alllll the state and extra dataclasses